client/my-sites/activity/activity-log/threat-alert.jsx

```diff
@@ -32,129 +37,304 @@ import './threat-alert.scss';
+					{
+						components: {
+							themeSlug: (
+								<span className="activity-log__threat-alert-slug">{ extension.slug }</span>
+							),
+							version: (
+								<span className="activity-log__threat-alert-version">{ extension.version }</span>
+							),
+						},
+					}
+				);
+
+			case 'database':
+				return translate(
+					'Jetpack identified $(threatCount)d threat in your database.',
+					'Jetpack identified %(threatCount)d threats in your database.',
+					{
+						count: Object.keys( threat.rows ).length,
+						args: {
+							threatCount: Object.keys( threat.rows ).length,
+						},
+					}
+				);
+
+			case 'none':
+			default:
+				return translate( 'Threat found' );
+		}
 	}
-};
-
-const headerSubtitle = ( translate, threat ) => {
-	switch ( detailType( threat ) ) {
-		case 'core':
-			return translate( 'Vulnerability found in WordPress' );
-
-		case 'file':
-			return translate( 'Threat found ({{signature/}})', {
-				components: {
-					signature: (
-						<span className="activity-log__threat-alert-signature">{ threat.signature }</span>
-					),
-				},
-			} );
-
-		case 'plugin':
-			return translate( 'Vulnerability found in plugin' );
-
-		case 'theme':
-			return translate( 'Vulnerability found in theme' );
-
-		case 'none':
-		default:
-			return translate( 'Miscellaneous vulnerability' );
+
+	renderSubtitle() {
+		const { threat, translate } = this.props;
+
+		switch ( this.getDetailType( threat ) ) {
+			case 'core':
+				return translate( 'Vulnerability found in WordPress' );
+
+			case 'file':
+				return translate( 'Threat found ({{signature/}})', {
+					components: {
+						signature: (
+							<span className="activity-log__threat-alert-signature">{ threat.signature }</span>
+						),
+					},
+				} );
+
+			case 'plugin':
+				return translate( 'Vulnerability found in plugin' );
+
+			case 'theme':
+				return translate( 'Vulnerability found in theme' );
+
+			case 'database':
+				return null;
+
+			case 'none':
+			default:
+				return translate( 'Miscellaneous vulnerability' );
+		}
 	}
-};
 
-export class ThreatAlert extends Component {
-	state = { requesting: false };
+	getGroupedThreatRows() {
+		const {
+			threat: { rows },
+			siteSlug,
+		} = this.props;
+		let infectedPosts = [];
 
-	handleFix = () => {
-		this.setState( { requesting: true } );
-		this.props.fixThreatAlert( this.props.siteId, this.props.threat.id );
-	};
+		function findObjectIndexInArray( array, attr, value ) {
+			for ( var i = 0; i < array.length; i++ ) {
+				if ( array[ i ][ attr ] === value ) {
+					return i;
+				}
+			}
+			return -1;
+		}
 
-	handleIgnore = () => {
-		this.setState( { requesting: true } );
-		this.props.ignoreThreatAlert( this.props.siteId, this.props.threat.id );
-	};
+		Object.keys( rows ).map( idx => {
+			const row = rows[ idx ];
+			const postIndex = findObjectIndexInArray( infectedPosts, 'postTitle', row.description );
```

Comment:
Nothing wrong with this, but could use lodash `find` to return the post instead, then manipulate it directly.